sdk/provisioning/Azure.Provisioning.CloudMachine/src/OFX/StorageServices.cs

```diff
@@ -85,16 +85,16 @@ private BlobClient GetBlobClientFromPath(string path, string? containerName)
 
     private static string ConvertPathToBlobPath(string path, BlobContainerClient container)
     {
-        if (Uri.TryCreate(path, UriKind.Absolute, out Uri blobUri))
+        if (Uri.TryCreate(path, UriKind.Absolute, out Uri? blobUri))
         {
             if (blobUri.Host == container.Uri.Host)
                 return blobUri.AbsoluteUri.Substring(container.Uri.AbsoluteUri.Length);
             if (!string.IsNullOrEmpty(blobUri.LocalPath))
             {
-                return blobUri.LocalPath.Substring(Path.GetPathRoot(path).Length).Replace('\\', '/');
+                return blobUri.LocalPath.Substring(Path.GetPathRoot(path)?.Length ?? 0).Replace('\\', '/');
```

Comment:
What's happening here and will it work correctly if Path.GetPathRoot(path) is null?